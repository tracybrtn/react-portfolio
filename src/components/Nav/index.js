//import React
import React from "react";

//Nav component function
function Nav(props) {
  //pass properties
  const {
    currentSection,
    setCurrentSection
  } = props;

  //return JSX
  return(
      <nav>
        <ul className="d-flex flex-row">
          <li className={currentSection === "about" ? "navActive" : "mx-2"}>
            <span onClick={() => setCurrentSection("about")}>About me</span>
          </li>
          <li className={currentSection === "portfolio" ? "mx-2 navActive" : "mx-2"}>
            <span onClick={() => setCurrentSection("portfolio")}>Portfolio</span>
          </li>
          <li className={currentSection === "contact" ? "mx-2 navActive" : "mx-2"}>
            <span onClick={() => setCurrentSection("contact")}>Contact</span>
          </li>
          <li className={currentSection === "resume" ? "mx-2 navActive" : "mx-2"}>
            <span onClick={() => setCurrentSection("resume")}>Resume</span>
          </li>
        </ul>
      </nav>
  );
}

//Export nav component
export default Nav;