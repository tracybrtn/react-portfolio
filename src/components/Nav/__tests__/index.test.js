//import dependencies
import React from "react";
import { render, cleanup } from '@testing-library/react';

import Nav from '..';

//clean up function
afterEach(cleanup);

const mockCurrentSection = jest.fn();
const mockSetCurrentSection = jest.fn();

describe('Nav component', () => {
  //baseline test
  it('renders', () => {
    render(<Nav 
      currentSection={mockCurrentSection}
      setCurrentSection={mockSetCurrentSection}
      />);
  })

  //snapshot test
  it('matches snapshot', () => {
    const { asFragment } = render(<Nav />);

    expect(asFragment()).toMatchSnapshot();
  });
});
