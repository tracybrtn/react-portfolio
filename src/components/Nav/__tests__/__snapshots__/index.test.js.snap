// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="d-flex flex-row"
    >
      <li
        class="mx-2"
      >
        <span>
          About me
        </span>
      </li>
      <li
        class="mx-2"
      >
        <span>
          Portfolio
        </span>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
